src/system.c

```diff
@@ -883,7 +883,7 @@ int open( char *file, int flags, int mode )
  
             void *ptr = fs->open( file + __strlen( filesystems[mapping].prefix ), flags );
 
-            if( ptr )
+            if( ptr && ((int32_t) ptr) != -2 )
```

Comment:
This seems to be a very specific magic value testing against a particular implementation of a filesystem and probably wouldn't work given any other circumstances. Did you track down why we're getting `-2` returned in the first place?